src/us/kbase/auth2/lib/Password.java

```diff
@@ -42,4 +76,42 @@ public void clear() {
 			password[i] = '0';
 		}
 	}
+	
+	/** Check for password validity (length and strength requirements)
+	 * @throws IllegalPasswordException if the password is not valid
+	 */
+	public void checkValidity() throws IllegalPasswordException {
+		// check length requirements
+		if(password.length > MAX_PASSWORD_LENGTH) {
+			throw new IllegalPasswordException("Password exceeds max length ("+
+												MAX_PASSWORD_LENGTH+")");
+		}
+		
+		// check strength requirement
+		final Strength strength = new Zxcvbn().measure(new String(password));
+		if(strength.getScore() < MIN_PASSWORD_STRENGTH_SCORE) {
+			String warning = strength.getFeedback().getWarning();
```

Comment:
Not that detects non-finals that are never changed.

 http://stackoverflow.com/questions/11398194/is-there-a-way-to-have-eclipse-automatically-make-all-my-variables-final